client/lib/form/post.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { forOwn } from 'lodash';
+
+export default function postForm( url, formData ) {
```

Comment:
I'm fine with this approach. If we keep this solution we should find better name for this method. My pick would be: `createFormAndSubmit`.